crates/sui/src/sui_commands.rs

```diff
@@ -1011,9 +1012,22 @@ async fn genesis(
     if client_config.active_address.is_none() {
         client_config.active_address = active_address;
     }
+
+    // On windows, using 0.0.0.0 will usually yield in an networking error. This localnet ip
+    // address must bind to 127.0.0.1 if the default 0.0.0.0 is used.
+    let localnet_ip =
+        if fullnode_config.json_rpc_address.ip() == IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)) {
+            "127.0.0.1".to_string()
```

Comment:
Is there somewhere we can change this more uniformly so that we don't introduce `0.0.0.0` in the config anymore?

Reply:
This is set in the genesis function, here. I don't like the solution either, but I do not know if I will break anything if I change the JSON RPC address there.

https://github.com/MystenLabs/sui/blob/main/crates/sui/src/sui_commands.rs#L960

Reply:
Yes, I think this does have broader implications: https://stackoverflow.com/questions/20778771/what-is-the-difference-between-0-0-0-0-127-0-0-1-and-localhost

TL;DR: `0.0.0.0` means accept traffic from all interfaces, whereas `127.0.0.1` means accept traffic from the loopback interface only (only traffic from the same host), conventionally.

This makes me wonder whether this is the right move even for Windows?

Reply:
Actually, maybe this is fine, and it suggests that changing it during environment creation is the right thing to do: We're saying that the config during genesis (which tells the network where to accept connections from) should accept from any interface, but when we are making the connection, we choose to make it via the loopback -- that seems fine.